### PR TITLE
fix(components/datetime): fix date range picker to emit `valueChanges` only once per change

### PR DESCRIPTION
**Pull request by Blackbaud-SteveBrush**
[AB#3101894](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/3101894)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2867/storybook/)

Component Storybooks:

- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/2867/storybooks/ag-grid-storybook/)
- [datetime-storybook](https://blackbaud.github.io/skyux-pr-preview/2867/storybooks/datetime-storybook/)
- [lists-storybook](https://blackbaud.github.io/skyux-pr-preview/2867/storybooks/lists-storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/2867/storybooks/tabs-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2867/code-examples/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2867/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2867/playground/)

File: libs/components/datetime/src/lib/modules/date-range-picker/date-range-picker.component.spec.ts [hunk 1]
```diff
@@ -210,7 +210,7 @@ describe('Date range picker', function () {
 
     detectChanges();
 
-    selectCalculator(SkyDateRangeCalculatorId.Before);
+    selectCalculator(SkyDateRangeCalculatorId.After);
```
Comment (Blackbaud-SteveBrush) [inline libs/components/datetime/src/lib/modules/date-range-picker/date-range-picker.component.spec.ts:213]: This test is checking the start date value, but the `Before` calculator only interacts with the end date. Fixed it.

File: libs/components/datetime/src/lib/modules/datepicker/datepicker.component.ts [hunk 2]
```diff
@@ -108,6 +109,12 @@ export class SkyDatepickerComponent implements OnDestroy, OnInit {
   @Output()
   public openChange = new EventEmitter<boolean>();
 
+  /**
+   * Fires when a user selects a date from the calendar.
+   * @internal
+   */
+  public calendarDateChange = output<Date>();
```
Comment (Blackbaud-SteveBrush) [inline libs/components/datetime/src/lib/modules/datepicker/datepicker.component.ts:116]: I need this to respond to user interactions, not arbitrary value changes.

File: libs/components/datetime/src/lib/modules/date-range-picker/date-range-picker.component.ts [hunk 3]
```diff
@@ -249,15 +264,15 @@ export class SkyDateRangePickerComponent
...
+  protected endDateHasErrors = signal<boolean>(false);
   protected formGroup: FormGroup;
   protected hostControl: AbstractControl | null | undefined;
   protected selectedCalculator: SkyDateRangeCalculator;
-  protected showEndDatePicker = false;
-  protected showStartDatePicker = false;
-  protected startDateHasErrors = false;
+  protected showEndDatePicker = signal<boolean>(false);
+  protected showStartDatePicker = signal<boolean>(false);
+  protected startDateHasErrors = signal<boolean>(false);
```
Comment (Blackbaud-SteveBrush) [inline libs/components/datetime/src/lib/modules/date-range-picker/date-range-picker.component.ts]: Using signals to avoid calls to `markForCheck()`.

**Review by Blackbaud-CoreyArcher — APPROVED** [review]

**Pull request merged**